### ISSUE TITLE
Sengoku Musou 3 Z Special: Popup error dialog box when select custom character

### ISSUE THREAD

**Issue by leoxxx** [hrydgard/ppsspp#12512]
As shown in the picture.
![1](https://user-images.githubusercontent.com/5364709/70850316-d3daa880-1ec3-11ea-8f8b-1f89e7638521.gif)

Not error in blank unit. If you watched a existing unit, then move up  another  existing unit, it pop up the error dialog box .

> ---------------------------
> Critical
> ---------------------------
> Can't have both THROUGH and USE_HW_TRANSFORM together!
> 
> Try to continue?
> ---------------------------
> 是(Y)   否(N)   
> ---------------------------

It will force close when you select Y or N.

https://img.vim-cn.com/89/eb73da81a019749e64576813803a1bb6cf8283.gif

**Comment by unknownbrackets** [comment]
Just to be 100% sure: if you clear out your cache (on Android, the app cache by long pressing on it, on Windows the PSP/SYSTEM/CACHE folder), does this still happen?

This is very interesting because we want to find the cause of this error, but it can also be caused by cached shaders.

Another interesting thing would be to create a save state right before the error, and then trigger the error.  Does it happen if you load the save state right after loading the game?  This would confirm if it has to do with other shaders - if possible, clear the cache before reloading the game too.

-[Unknown]

**Comment by leoxxx** [comment]
Try to clear the cache, don't popup the error dialog, but it freeze.
Another interesting thing, if you move the selection box fast enough (no display character model ), it don't triger!

**Comment by unknownbrackets** [comment]
It's very interesting that it doesn't trigger in some cases.  It could still be related to #9993 - that indicates somehow the game gets into bad display list ram or misunderstands a BJUMP command.

If you enable software rendering (it will be very slow), does it freeze/crash?  I recommend using a save state to get near it.

Also, it might help if you can upload your save state here.

-[Unknown]

**Comment by leoxxx** [comment]
[PPSSPP_STATE.zip](https://github.com/hrydgard/ppsspp/files/4014048/PPSSPP_STATE.zip)
0=disable software rendering.
1=enable software rendering.
0. Now it must be frozen the game, if I want trigger the diaglog. But ppsspp is O.K.
1. In this mode, the game lag sometimes. The map have some error and it is blur.  It must be force close ppsspp.

**Comment by RobertSVG15** [comment]
Hello! The same error occurs in different parts of the game menus.

When I try to change an outfit from the DLCs; the same happens.

You can find the DLCs here https://cdromance.com/sony-psp-dlc-list-psp-downloadable-content/

Please, help

**Comment by RobertSVG15** [comment]
@hrydgard please take a look to this issue, it's been a while (around 4 years) since this error was found, and until now nobody has checked into it. I don't know how to, so I please you for your help in this matter. Once again, please don't forget.

Thank you.

**Comment by RobertSVG15** [comment]
**Update:**

**1)** The error reported by @leoxxx **can be avoided** by removing the other created characters and leaving only 1 of them (the one to be used). But of course, that's only a partial solution.

![A](https://user-images.githubusercontent.com/59510411/77257153-f6115e80-6c48-11ea-9032-80b72587eaa7.jpg)

**2)** The same type of critical error occurs when trying to change the costume of an official character using DLCs in the Story Mode and Free Mode menu.

This is the menu. If you have the DLCs installed and press square; you access in the costume change menu:

![B](https://user-images.githubusercontent.com/59510411/77257170-0fb2a600-6c49-11ea-9a9c-509311a7f4e6.jpg)

![C](https://user-images.githubusercontent.com/59510411/77257214-50aaba80-6c49-11ea-92ec-dea75e5d48f6.jpg)

But, if you press R or L to change the costume, PPSSPP stops working or the game stops working and then PPSSPP does.

![D](https://user-images.githubusercontent.com/59510411/77257268-a41d0880-6c49-11ea-9f57-6fcd29bc0d66.jpg)

It reads:

"PPSSPP has stopped working"
"Restart the application"

**3)** The same error occurs randomly in certain areas of the game menus.

The rest of the game runs perfectly! I've explained my device's system in [#12742](https://github.com/hrydgard/ppsspp/issues/12742)

**Comment by unknownbrackets** [comment]
Interesting.  Does that crash still occur if you disable fast memory?  That isn't a fix (and indicates it's still working incorrectly), but it would help us understand the crash better.

-[Unknown]

**Comment by RobertSVG15** [comment]
> Interesting. Does that crash still occur if you disable fast memory? That isn't a fix (and indicates it's still working incorrectly), but it would help us understand the crash better.
> 
> -[Unknown]

Yes, it does

**Comment by leoxxx** [comment]
> 
> 
> **Update:**
> 
> **1)** The error reported by @leoxxx **can be avoided** by removing the other created characters and leaving only 1 of them (the one to be used). But of course, that's only a partial solution.
> 
> ![A](https://user-images.githubusercontent.com/59510411/77257153-f6115e80-6c48-11ea-9032-80b72587eaa7.jpg)
> 
> **2)** The same type of critical error occurs when trying to change the costume of an official character using DLCs in the Story Mode and Free Mode menu.
> 
> This is the menu. If you have the DLCs installed and press square; you access in the costume change menu:
> 
> ![B](https://user-images.githubusercontent.com/59510411/77257170-0fb2a600-6c49-11ea-9a9c-509311a7f4e6.jpg)
> 
> ![C](https://user-images.githubusercontent.com/59510411/77257214-50aaba80-6c49-11ea-92ec-dea75e5d48f6.jpg)
> 
> But, if you press R or L to change the costume, PPSSPP stops working or the game stops working and then PPSSPP does.
> 
> ![D](https://user-images.githubusercontent.com/59510411/77257268-a41d0880-6c49-11ea-9f57-6fcd29bc0d66.jpg)
> 
> It reads:
> 
> "PPSSPP has stopped working"
> "Restart the application"
> 
> **3)** The same error occurs randomly in certain areas of the game menus.
> 
> The rest of the game runs perfectly! I've explained my device's system in [#12742](https://github.com/hrydgard/ppsspp/issues/12742)

> Another interesting thing, if you move the selection box fast enough (no display character model ), it don't trigger!

**Comment by RobertSVG15** [comment]
Hi, I found a way to use all 16 slots.  You must go to Samurai Residence and look for the menu that has a rooster.  There you install the Game Data (it looks invisible went you go to this option).  By doing that I managed to avoid the bug.  However, the problem persists with DLCs, it bothers me.  

I found that the problem indeed has to do with the inserted stick memory and the frames, I have tried several options but none worked.  The PPSSPP memory options seem to help but do not solve the problem.

@leoxxx 

I just want to fix the DLC issue, but I can't... At least by myself.

**Comment by leoxxx** [comment]
It's not freeze in v1.10, but still have error.
Black screen in free mode and green screen in history mode, but still can push the button. After mission start, it will be fine.

**Comment by RobertSVG15** [comment]
@hrydgard hello! This game has been working better after all this time.

But, there is only one big problem about it.

When I try to use the DLC to change the officer's outfits; the game crashes.

Please help, any idea about what's going on? What should I do?

**Comment by hrydgard** [comment]
Hard to say, probably no way to work around it right now. Might be some emulator bug with DLC handling, I don't know..

**Comment by IrfanH495** [comment]
The DLC works well, but if I choose a costume too long it will freeze to 0/0 100 fps.
https://drive.google.com/file/d/11T5TRzlhd1xjOdP0le-oCIKzZ52zDh-h/view?usp=drivesdk
I added it in compat.ini
[BlockTransferAllowCreateFB]
# 戦国無双３ Z Special
ULJM06024 = true
[IntraVRAMBlockTransferAllowCreateFB]
# 戦国無双３ Z Special
ULJM06024 = true[ForceUMDDelay]
# 戦国無双３ Z Special
ULJM06024 = true
what else should be added to fix it.?

**Comment by IrfanH495** [comment]
I don't know which to add or remove from the list above.
please check it

**Comment by leoxxx** [comment]
PPSSPP\assets\compat.ini ?

**Comment by IrfanH495** [comment]
> PPSSPP\assets\compat.ini ?

yes

**Comment by IrfanH495** [comment]
change costumes at the time of the fan image.  if you don't want fc I've tried it it worked
![Screenshot_20211224-092252](https://user-images.githubusercontent.com/78854859/147308044-8bcda367-df11-465f-83ad-2973e1f7f502.png)
If it's too late press x/o If you don't want fc
![Screenshot_20211224-091953](https://user-images.githubusercontent.com/78854859/147308166-e3bac3e8-d7a9-4151-9081-cf7ce204e89c.png)

**Comment by IrfanH495** [comment]
![2021_12_24_10_02_35](https://user-images.githubusercontent.com/78854859/147310863-ff343432-8220-425a-9fd8-453ff1a50656.gif)

**Comment by IrfanH495** [comment]
> PPSSPP\assets\compat.ini ?

i don't know if on pc, i play on android PSP/SYSTEM/compat.ini

**Comment by IrfanH495** [comment]
> As shown in the picture. ![1](https://user-images.githubusercontent.com/5364709/70850316-d3daa880-1ec3-11ea-8f8b-1f89e7638521.gif)
> 
> Not error in blank unit. If you watched a existing unit, then move up another existing unit, it pop up the error dialog box .
> 
> > ## Critical
> > Can't have both THROUGH and USE_HW_TRANSFORM together!
> > ## Try to continue?
> > ## 是(Y)   否(N)
> 
> It will force close when you select Y or N.
> 
> https://img.vim-cn.com/89/eb73da81a019749e64576813803a1bb6cf8283.gif

like in this video there is a fan image before loading the character
https://youtu.be/Cw3c8q7WMWQ

**Comment by unknownbrackets** [comment]
It sounds like the original issue here is potentially fixed, but there may still be an issue (potentially related, potentially completely separate.)

The new issue happens when you wait on that screen for a while.  If you confirm the costume quickly, it doesn't crash.  Is that correct?

Have you tried switching backends (i.e. Vulkan vs OpenGL) in settings?

-[Unknown]

**Comment by IrfanH495** [comment]
i have tried both opengl & vulkan both work for me
opngl
https://drive.google.com/file/d/12Gy_GpzREnM6oSNxCQIwd3gE5mACw5p5/view?usp=drivesdk
vulkan
https://drive.google.com/file/d/12FqC8G7z0XWpOXXN-XPqP128fXVpPRF5/view?usp=drivesdk
wait it's not finished, there's still something I need to record again

**Comment by IrfanH495** [comment]
> As shown in the picture. ![1](https://user-images.githubusercontent.com/5364709/70850316-d3daa880-1ec3-11ea-8f8b-1f89e7638521.gif)
> 
> Not error in blank unit. If you watched a existing unit, then move up another existing unit, it pop up the error dialog box .
> 
> > ## Critical
> > Can't have both THROUGH and USE_HW_TRANSFORM together!
> > ## Try to continue?
> > ## 是(Y)   否(N)
> 
> It will force close when you select Y or N.
> 
> https://img.vim-cn.com/89/eb73da81a019749e64576813803a1bb6cf8283.gif

as reported by @leoxxx above
it gets better even after loading the character it's not fc
vulkan 
https://drive.google.com/file/d/12V-6k8JlnxPCz3lEyaEcHe61ma6hZxiC/view?usp=drivesdk
opengl
https://drive.google.com/file/d/12ZpAaQr4XygAHEU5epKJexRvCLTeRePj/view?usp=drivesdk

I just added it in compat.ini
[ForceUMDDelay]
戦国無双３ Z Special
ULJM06024 = true
if force change character after loading it sometimes fc I don't know maybe my device is not very strong
![Screenshot_20211224-091953](https://user-images.githubusercontent.com/78854859/147379583-d89362da-f5e5-4f17-9be6-da7227a29cbd.png)

**Comment by kotarofumafan** [comment]
Hello everyone i'm on a phone and i want to have DLC sengoku musou 3z i have put the file ULJM on the file psp/game but i can't play with DLC please help me . sorry for m'y poor english i'm french

**Comment by kotarofumafan** [comment]
And same for musou orochi 2

**Comment by IrfanH495** [comment]
> Hello everyone i'm on a phone and i want to have DLC sengoku musou 3z i have put the file ULJM on the file psp/game but i can't play with DLC please help me . sorry for m'y poor english i'm french

try this one
https://youtu.be/g3ZBYmjUo-Y

**Comment by Zh4ngl1ao** [comment]
Hello everybody i have a problem with the dlc of Sw3z i extract the file ULJM on PSP/games but when i go on story mode or free i don't have the square touch please help me

**Comment by Zh4ngl1ao** [comment]
I have the square touch i take the app Zarchiver and it's good ^^

**Comment by IrfanH495** [comment]
[v1.13.2-1944-g3e481634a]
i set cpu clock to 38 it doesn't experience fc and freezes but fps drops 30/30, 25/25, 23/23.
if i use compact ForceUM Delay ULJM06024 = true and cpu clock 38. fps is better 30/30, 28/28.  the game doesn't freeze /fc.
I didn't make that video, but it works fine both vulkan and opengl

**Comment by leoxxx** [comment]
Setting CPU to 38HZ is O.K. But more lag.

**Comment by leoxxx** [comment]
Setting CPU to 38HZ is O.K. But more lag.

**Comment by IrfanH495** [comment]
@leoxxx sorry I didn't know when the game started it would be like this
![Screenshot_20221119-201023_PPSSPP](https://user-images.githubusercontent.com/78854859/202852452-363ee450-a3ae-4d51-9048-777178e3fd0e.png)

**Comment by IrfanH495** [comment]
ppsspp-v1.13.2-1998-g5c2663e37-android
black screen, after changing costume.
not using compact , all default settings
GE frame 
[ULJM06024.zip](https://github.com/hrydgard/ppsspp/files/10068803/ULJM06024.zip)

**Comment by unknownbrackets** [comment]
Something (a cheat?) has executed a corrupted display list here (which is about what I'd expect from the previous notes about a crash.)  It has the logic op flag enabled, but the enable is garbage - 28323235.  This probably means one of:
 * A cheat broke things.
 * A GE block transfer overwrote memory.
 * A GE list writing cache/timing problem.

Without the corruption, it looks like this:
![image](https://user-images.githubusercontent.com/191233/203679348-aebe3743-f40f-4f63-934a-2dd00b7156d6.png)

This is what I forced off to make it look like that, just for future reference:
```c++
	if ((gstate.cullfaceEnable & 0x00FFFFFF) == 0x00191414)
		gstate.cullfaceEnable &= 0xFF000000;
	if ((gstate.fogEnable & 0x00FFFFFF) == 0x001c1c14)
		gstate.fogEnable &= 0xFF000000;
	if ((gstate.stencilTestEnable & 0x00FFFFFF) == 0x001d1d19)
		gstate.stencilTestEnable &= 0xFF000000;
	if ((gstate.logicOpEnable & 0x00FFFFFF) == 0x00323235)
		gstate.logicOpEnable &= 0xFF000000;
	if ((gstate.texmapmode & 0x00FFFFFF) == 0x00b31b62)
		gstate.texmapmode &= 0xFF000000;
```

Is that how you'd expect it to work?

Is it possible to get to this same screen using only save data (no save state) and no cheats?  Does it still happen then?

If yes, it'd help to also try using the software renderer.  Does it also happen when using the software renderer?

-[Unknown]

**Comment by IrfanH495** [comment]
no cheat, all default setting, no save state,No save data
opengl games freeze
Vulkan games freeze
software renderer games freeze
![Screenshot_20221124-125731_PPSSPP](https://user-images.githubusercontent.com/78854859/203706224-f7f8760e-3bd7-4473-ae1b-80756ab70c86.png)
![Screenshot_20221124-130422_PPSSPP](https://user-images.githubusercontent.com/78854859/203706478-c1b02a65-8433-4f16-85bc-cb1a0572078d.png)
![Screenshot_20221124-130452_PPSSPP](https://user-images.githubusercontent.com/78854859/203706497-86e84721-2f9e-415a-a036-661579deeebc.png)
opengl & vulkan 2xpsp
![Screenshot_20221124-130644_PPSSPP](https://user-images.githubusercontent.com/78854859/203706932-062918a2-1d9e-4d75-8538-b225baa6545f.png)
![Screenshot_20221124-130655_PPSSPP](https://user-images.githubusercontent.com/78854859/203706961-f4e77a52-f45b-426e-abec-0c3082f5a827.png)
![Screenshot_20221124-130701_PPSSPP](https://user-images.githubusercontent.com/78854859/203706973-d67762c4-1ee0-49b0-bfbe-b806b96d4e3c.png)

**Comment by IrfanH495** [comment]
> Is it possible to get to this same screen using only save data (no save state) and no cheats? Does it still happen then?

Impossible to get custom without dlc.
there is a picture that is almost similar only this
![Screenshot_20230630-042421_PPSSPP](https://github.com/hrydgard/ppsspp/assets/78854859/9bedfb54-172d-48a7-9601-5d81ff67369c)
the screen turns white or blue, if used software renderer
No cheat,No DLC, No Save State,no save data
![Screenshot_20230630-043000_PPSSPP](https://github.com/hrydgard/ppsspp/assets/78854859/d83117d9-aafc-418f-8dd5-087a912e9f21)

GE frame, software renderer
[ULJM06024 (1) Sengoku mosou 3z.zip](https://github.com/hrydgard/ppsspp/files/11910711/ULJM06024.1.Sengoku.mosou.3z.zip)
